version: "3.9"
services:
  postgres:
    image: postgres:14.8-alpine
    ports:
      - "5435:5432"
    restart: always
    volumes:
      - postgres:/var/lib/postgresql/data/
      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5
  django:
    build: ./django
    env_file:
      - .env
    ports:
      - "8000:8000"
    healthcheck:
      test: "curl --fail http://localhost:8000/ || exit 1"
      interval: 20s
      retries: 5
      timeout: 10s
    depends_on:
      postgres:
        condition: service_healthy
  sensor:
    build: ./sensor
    env_file:
      - .env
    depends_on:
      django:
        condition: service_healthy
  web:
    build: 
      context: ./web
    links:
      - "django:api"
volumes:
  postgres:
    driver: local